docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,200 @@
+
+![multi-user v. multi tenant](multi-user v multi-tenant.png)
+
+This *conceptual difference* is often confused with the *implementation difference*. It is perfectly possible for the provider of a multi-tenanted service to implement their service using a single set of resources shared between all their tenants (if other constraints permit).
+
+## When should a solution be multi-tenanted, not multi-user?
+A multi-tenanted solution is required when one or more groups of users have a strong requirement for *isolation* from other groups of users. This is not simply a secure connection to services for that tenant. Isolation will typically manifest in one or more of the following ways:
+
+1. The ability for a tenant to provision, manage, and monitor a collection of resources independently of other tenants.
+1. The ability for a tenant to bring their own resources (including, for example, domain names, application code, configuration, data, cryptographic keys, policy) and integrate into their solution
+1. The ability to provide particular kinds of security boundary around their solution and control identity and access (including the ability to deny access to the entity that provides the tenant with those resources)
+1. The inability for one tenant to determine the existence or otherwise of another tenant.
+1. The inability for one tenant to interfere with the operation or performance of another tenant e.g. by starving it of resources.
+
+Note that this is not simply *data isolation*. Most data storage solutions (blob storage, SQL databases etc.) have mechanisms that allow you to control access to subsets of the data, constrained to particular identities. For example, techniques like Row Level Security (RLS), Attribute Based Access Control (ABAC), filtering, and data partitioning provide various types of security and access boundary. Multi-user systems will often use these techniques to ensure segregation of user data.
+
+A multi-tenanted system will also use these techniques to provide data segregation, but adds in other constraints from the list above. 
+
+## Tenancy as a hierarchy
+It is also very common for a multi-tenanted solution to itself be used to provide other multi-tenanted solutions. This can be a further source of confusion.
+
+For example, the Microsoft Azure platform is a multi-tenanted solution. 
+
+An Azure tenant has resources that are isolated in all the ways described above, and we can use those resources to build solutions which are themselves multi-tenanted.
+
+With that in mind, you can think of tenancy as a tree, with a *root tenant* which owns the multi-tenanted solution. This root tenant has zero or more *child tenants* for which it provides some (other) services.
+
+![Tenant hierarchies](Tenant hierarchies.png)
```

Comment:
This diagram (aside from not rendering due to spaces in the link and no `<>`) shows a hierarchy of homogenous tenants. So this illustrates only one part of the picture, and doesn't really incorporate the idea you just discussed that this multi-tenanted solution might be supported by a different multi-tenanted solution (e.g., Azure) underneath.

While that might be the right thing (a diagram showing both might be quite confusing) I don't think it's well-signalled that this diagram doesn't show the things you were talking about just moments ago at the start of this section.